packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {Helmet} from 'react-helmet'
+import {isServer} from '@tanstack/react-query'
+
+/** Origins that are allowed to run Storefront Preview. */
+const TRUSTED_ORIGINS = [
+    'https://runtime.commercecloud.com',
+    'https://runtime-admin-staging.mobify-storefront.com',
+    'https://runtime-admin-preview.mobify-storefront.com'
+]
+
+/** Origin used for local Runtime Admin. */
+const DEVELOPMENT_ORIGIN = 'http://localhost:4000'
+
+/** Detects whether the storefront is running in an iframe. */
+const detectInIframe = () => typeof window !== 'undefined' && window.parent !== window.self
+
+/** Gets the parent origin when running in an iframe. */
+const getParentOrigin = () => {
+    if (detectInIframe()) {
+        if (window.location.ancestorOrigins) return window.location.ancestorOrigins[0]
+        // ancestorOrigins does not exist in Firefox, so we use referrer as a fallback
+        if (document.referrer) return new URL(document.referrer).origin
+    }
+}
+
+const isParentOriginTrusted = (parentOrigin) => {
+    return window.location.hostname === 'localhost'
+        ? parentOrigin === DEVELOPMENT_ORIGIN // Development
+        : TRUSTED_ORIGINS.includes(parentOrigin) // Production
+}
+
+/** Detects whether the storefront is running in an iframe as part of Storefront Preview. */
+export const detectStorefrontPreview = () => {
+    const parentOrigin = getParentOrigin()
+    return Boolean(parentOrigin) && isParentOriginTrusted(parentOrigin)
+}
+
+/** Returns the URL to load the Storefront Preview client script from the parent origin. */
+export const getClientScript = () => {
+    const parentOrigin = getParentOrigin() ?? 'https://runtime.commercecloud.com'
+    return parentOrigin === DEVELOPMENT_ORIGIN
+        ? `${parentOrigin}/mobify/bundle/development/static/storefront-preview.js`
+        : `${parentOrigin}/cc/b2c/preview/preview.client.js`
+}
+export const StorefrontPreview = ({
+    enabled = typeof window !== 'undefined' ? window.STOREFRONT_PREVIEW.enabled : false,
+    customisation
+}) => {
+    let isHostTrusted
+    useEffect(() => {
+        if (enabled && isHostTrusted) {
+            window.STOREFRONT_PREVIEW = {
+                ...window.STOREFRONT_PREVIEW,
+                ...customisation
+            }
+        }
+    }, [enabled])
+    if (!enabled) {
+        return null
+    }
+    isHostTrusted = detectStorefrontPreview()
```

Comment:
```suggestion
    if (!enabled) {
        return null
    }
    const isHostTrusted = detectStorefrontPreview()
    useEffect(() => {
        if (enabled && isHostTrusted) {
            window.STOREFRONT_PREVIEW = {
                ...window.STOREFRONT_PREVIEW,
                ...customisation
            }
        }
    }, [enabled])
```

Reply:
This would break React! Can't conditionally return before using a hook.

Reply:
yea, as Will said, linting will not allow conditional return before using hook. Hook will need to run every render.